internal/cmd/eventgen/templates/eventfile.tmpl

```diff
@@ -19,9 +19,26 @@
 package eiffelevents
 
 import (
+	"time"
+
 	"github.com/clarketm/json"
+	"github.com/google/uuid"
 )
 
+// New{{.StructName}} creates a new struct pointer that represents
+// major version {{.MajorVersion}} of {{.EventType}}.
+// The returned struct has all required meta members populated.
+// The event version is set to the most recent {{.MajorVersion}}.x.x
+// currently known by this SDK.
+func New{{.StructName}}() (*{{.StructName}}, error) {
+	var event {{.StructName}}
+	event.Meta.Type = {{ printf "%q" .EventType }}
+	event.Meta.ID = uuid.NewString()
```

Comment:
A more general question: Should this field be of type uuid.UUID instead of sting in order to force a valid ID? Or should this be left up to the user to define by themselves.

Reply:
I've been debating this with myself a few times. The problem with making the meta.id field a "real" UUID is that an event with an invalid UUID can't be deserialized at all, and I'm not sure that's desirable behavior. When working with an event there's no point in having a specific data type since meta.id is always treated as an opaque value that you just pass around or maybe include in a log message. If the id isn't a valid UUID it'll get caught by the JSON validation. Well, maybe the regexp-based validation won't catch _all_ kinds of invalid UUIDs.

The same isn't true for meta.time since you might want to interpret the value and, say, compute the age of the event. So that field might change from a plain integer to something based on time.Time.